src/amo/containers/CategoryList.js

```diff
@@ -0,0 +1,30 @@
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+import { connect } from 'react-redux';
+
+import Categories from 'amo/components/Categories';
+import { loadCategoriesIfNeeded } from 'core/utils';
+
+
+export function mapStateToProps(state, ownProps) {
+  const addonType = ownProps.params.addonType.replace(/s$/, '');
+  const clientApp = state.api.clientApp;
+  const categories = state.categories.categories[clientApp][addonType] ?
```

Comment:
I definitely like that idea, but think it'd be something we should do later. Thanks for adding the comment though, the proposed idea is much nicer.